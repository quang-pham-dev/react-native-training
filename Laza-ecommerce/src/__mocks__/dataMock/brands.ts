/* A data array. */

export const brand = {
  id: '0',
  name: 'Nike',
  logoUrl:
    'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
};

export const brands = [
  {
    id: '0',
    name: 'Nike',
    logoUrl:
      'https://toppng.com/uploads/preview/ike-logo-png-hd-nike-logo-png-image-free-download-nike-logo-sv-11562865034fkhqkcoas6.png'
  },
  {
    id: '1',
    name: 'Adidas',
    logoUrl:
      'https://www.pikpng.com/pngl/m/42-429866_adidas-logo-png-images-free-download-rh-pngimg.png'
  },
  {
    id: '2',
    name: 'Fila',
    logoUrl:
      'https://toppng.com/uploads/preview/fila-logo-vector-download-free-11574216707jcmvfu58fy.png'
  },
  {
    id: '3',
    name: 'Puma',
    logoUrl:
      'https://banner2.cleanpng.com/20180613/gsj/kisspng-tracksuit-puma-logo-brand-clothing-puma-logo-5b20dc6b232426.487874081528880235144.jpg'
  },
  {
    id: '4',
    name: 'Gucci',
    logoUrl: 'https://esquilo.io/png/thumb/XBwczcMncYeff65-Vector-Gucci-Logo-PNG-Free-Download.png'
  }
];
