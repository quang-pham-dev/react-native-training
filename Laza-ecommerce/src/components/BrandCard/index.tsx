import React, { memo, useCallback } from 'react';
import { Image, TouchableOpacity, View } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Title from 'components/Title';

// Types
import { IBrandCardProps } from 'types/models/Brands';

// Styles
import styles from './styles';

const BrandCard = ({ brand, onPressBrandCard, brandCardStyles }: IBrandCardProps) => {
  // handle action press card brand
  const handlePressBrandCard = useCallback(() => {
    onPressBrandCard(brand.id);
  }, [brand.id, onPressBrandCard]);

  return (
    <TouchableOpacity
      testID='brandCard'
      onPress={handlePressBrandCard}
      style={[styles.brandCardContainer, brandCardStyles]}>
      <View style={[styles.brandCardWrapper]}>
        <View style={styles.brandLogoWrapper}>
          <Image style={styles.brandLogo} source={{ uri: brand.logoUrl }} />
        </View>
        <Title titleName={brand.name} />
      </View>
    </TouchableOpacity>
  );
};

export default memo(BrandCard, isEqual);
