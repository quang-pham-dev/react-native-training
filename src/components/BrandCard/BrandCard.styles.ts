import styled from 'styled-components/native'

// Libs
import {s} from 'react-native-size-matters/extend'

// Themes
import {Colors, Metrics} from '@themes'

export const BrandCardWrapperStyled = styled.TouchableOpacity`
  height: ${s(50)}px;
  margin-right: ${Metrics.margin.small}px;
  background-color: ${Colors.palette.lightGray};
  border-radius: ${Metrics.borderRadius.base}px;
  padding-right: ${Metrics.padding.medium}px;
`
export const ImageWrapperStyled = styled.View`
  height: ${s(40)}px;
  width: ${s(40)}px;
  justify-content: center;
  align-items: center;
  padding-vertical: ${Metrics.padding.xs}px;
  padding-horizontal: ${Metrics.padding.xs}px;
  margin-left: ${Metrics.margin.tiny}px;
  margin-right: ${Metrics.margin.small}px;
  margin-vertical: ${Metrics.margin.tiny}px;
  border-radius: ${Metrics.borderRadius.base}px;
  background-color: ${Colors.background};
`
