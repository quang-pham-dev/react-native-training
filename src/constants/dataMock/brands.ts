export const brand = {
  id: '0',
  name: 'Nike',
  url: 'https://1000logos.net/wp-content/uploads/2021/11/Nike-Logo.png',
}

export const brands = [
  {
    id: '0',
    name: 'Nike',
    url: 'https://1000logos.net/wp-content/uploads/2021/11/Nike-Logo.png',
  },
  {
    id: '1',
    name: 'Adidas',
    url: 'https://www.pikpng.com/pngl/m/42-429866_adidas-logo-png-images-free-download-rh-pngimg.png',
  },
  {
    id: '2',
    name: 'Fila',
    url: 'https://toppng.com/uploads/preview/fila-logo-vector-download-free-11574216707jcmvfu58fy.png',
  },
  {
    id: '3',
    name: 'Puma',
    url: 'https://banner2.cleanpng.com/20180613/gsj/kisspng-tracksuit-puma-logo-brand-clothing-puma-logo-5b20dc6b232426.487874081528880235144.jpg',
  },
  {
    id: '4',
    name: 'Gucci',
    url: 'https://esquilo.io/png/thumb/XBwczcMncYeff65-Vector-Gucci-Logo-PNG-Free-Download.png',
  },
]
