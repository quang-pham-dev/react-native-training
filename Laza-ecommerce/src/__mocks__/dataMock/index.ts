import { user } from './user';
import { brand, brands } from './brands';
import { product, products } from './products';

export { brand, brands, product, products, user };
