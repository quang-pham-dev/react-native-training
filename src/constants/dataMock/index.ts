import {user} from './user'
import {brand, brands} from './brands'

export {brand, brands, user}
