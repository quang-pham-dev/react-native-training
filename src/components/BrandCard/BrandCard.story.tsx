import React from 'react'

// Storybook
import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react-native'
import {StoryScreen, Story, UseCase} from '../../../storybook/views'

// Components
import BrandCard from './index'

// Constants
import {brand} from '@constants/dataMock'

declare let module: NodeModule

storiesOf('Brand Card', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Default', () => (
    <Story>
      <UseCase text="default">
        <BrandCard brand={brand} onPressBrandCard={action('clicked')} />
      </UseCase>
    </Story>
  ))
