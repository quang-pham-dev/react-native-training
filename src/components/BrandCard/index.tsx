import React, {memo, useCallback} from 'react'

// LIBS
import isEqual from 'react-fast-compare'
import FastImage from 'react-native-fast-image'
import {s} from 'react-native-size-matters/extend'

// Types
import {BrandCardProps} from '@model-types'

// Styles
import {BrandCardWrapperStyled, ImageWrapperStyled} from './BrandCard.styles'
import PStyled from '@components/Paragraph/P.styles'
import {Image} from '@components/Image/Image.styles'
import FlexStyled from '@components/Flex/Flex.styles'

// Constants
import {PARAGRAPH_TYPE} from '@constants'

const BrandCard = ({brand, onPressBrandCard}: BrandCardProps) => {
  const {id, name, url} = brand || {}
  // handle action press card brand
  const handlePressBrandCard = useCallback(() => {
    onPressBrandCard(id)
  }, [id, onPressBrandCard])

  return (
    <BrandCardWrapperStyled onPress={handlePressBrandCard}>
      <FlexStyled.RowCenterHorizontal>
        <ImageWrapperStyled>
          <Image.Normal
            alignSelf="center"
            height={s(40)}
            width={s(40)}
            source={{
              uri: url,
              priority: FastImage.priority.normal,
            }}
            resizeMode={FastImage.resizeMode.contain}
          />
        </ImageWrapperStyled>
        <PStyled.Base type={PARAGRAPH_TYPE.BRAND}>{name}</PStyled.Base>
      </FlexStyled.RowCenterHorizontal>
    </BrandCardWrapperStyled>
  )
}

export default memo(BrandCard, isEqual)
