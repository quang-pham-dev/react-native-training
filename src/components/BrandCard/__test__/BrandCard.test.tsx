import React from 'react'

// LIBS
import {render} from '@testing-library/react-native'

// Components
import BrandCard from '@components/BrandCard'

// Mocks
import {brand} from '@constants/dataMock'

describe('Brand Card Component', () => {
  const props = {
    brand: brand,
    onPressBrandCard: jest.fn(),
  }

  const tree = render(<BrandCard {...props} />)
  test('Should render correctly', () => {
    const component = tree.toJSON()
    expect(component).toMatchSnapshot()
  })
})
