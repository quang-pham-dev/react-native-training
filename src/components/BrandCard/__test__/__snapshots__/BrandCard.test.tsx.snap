// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand Card Component Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F5F6FA",
      "borderRadius": 10,
      "height": 50,
      "marginRight": 10,
      "opacity": 1,
      "paddingRight": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 5,
            "marginRight": 10,
            "marginVertical": 5,
            "paddingHorizontal": 6,
            "paddingVertical": 6,
            "width": 40,
          },
        ]
      }
    >
      <Image
        alignSelf="center"
        resizeMode="contain"
        source={
          Object {
            "uri": "https://1000logos.net/wp-content/uploads/2021/11/Nike-Logo.png",
          }
        }
        style={
          Array [
            Object {
              "alignSelf": "center",
              "height": "100%",
              "resizeMode": "contain",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#1D1E20",
            "fontFamily": "Inter-Medium",
            "fontSize": 15,
            "lineHeight": 16.5,
            "textAlign": "left",
          },
        ]
      }
      type="brand"
    >
      Nike
    </Text>
  </View>
</View>
`;
